### PR TITLE
a few bug fixed and updated zeroing logic

### PR DESCRIPTION
**Pull request by QuentinTorg**
should be faster now.... hopefully

### CODE REVIEW

File: Clock.ino [hunk 1]
```diff
@@ -27,6 +33,8 @@ struct Time
             Sec >= o.Sec && Min >= o.Mil;
     }
 
+
```
Comment (Nyarmith) [inline Clock.ino]: whitespace bloat

File: Clock.ino [hunk 2]
```diff
@@ -214,7 +222,10 @@ HourHand hourHand_;
 void setup()
 {
     initClockPins();
-    rtcClock_.init(globTime_);    
+    rtcClock_.init(globTime_); 
+
+    // send all axes to zero, then tell objects to reset to zero
+    zero_all(100, 1000);
```
Comment (Nyarmith) [inline Clock.ino]: this should be handled separately in the classes

File: ClockAPI.h [hunk 3]
```diff
@@ -71,33 +72,61 @@ void y_step(unsigned char dir)
...
+void zero_all(int fast_delay_micros, int slow_delay_micros)
 {
-  Serial.println("zeroing h");
-  while (digitalRead(LSWITCHH)) {
-    h_step(DIR_MINUS);
+  bool x_zeroed = false;
+  bool y_zeroed = false;
+  bool h_zeroed = false;
+  bool directions[] = {DIR_MINUS, DIR_PLUS, DIR_MINUS};
+  bool delays[] = {fast_delay_micros, fast_delay_micros, slow_delay_micros};
```
Comment (Nyarmith) [inline ClockAPI.h]: should delays be bool?

**Review by Nyarmith — CHANGES_REQUESTED** [review]
Motor control and zeroing are coupled, and the hour hand and gantry classes are orphaned from the actual motor interface. The motor interface should be independent and the speeds should be a separate enum coupled with that interface. Zeroing should be handled via the class abstractions, whose job is to couple the motor interface with other events like the switches.

File: Clock.ino (hunk 2 above)
Comment (QuentinTorg) [reply Clock.ino]: how should we do it at the same time within the classes? should there be a motor class so we can have motor objects and then you can zero all of the motors and then instantiate a gantry/hour hand with a reference to the motor objects? 

Then we also get away from having three different step functions. Step and direction are handled by the class when we initialize the motor. if one needs to be reversed, then reverse when its instantiated

File: ClockAPI.h (hunk 3 above)
Comment (QuentinTorg) [reply ClockAPI.h]: wise

File: Clock.ino (hunk 2 above)
Comment (QuentinTorg) [reply Clock.ino]: The Motor class would work in steps but the gantry or hour hand could work in millimeters or degrees

File: ClockAPI.h [hunk 4]
```diff
@@ -71,33 +71,64 @@ void y_step(unsigned char dir)
...
+void zero_all(int fast_delay_micros, int slow_delay_micros)
 {
-  Serial.println("zeroing h");
-  while (digitalRead(LSWITCHH)) {
-    h_step(DIR_MINUS);
+  bool x_zeroed = false;
+  bool y_zeroed = false;
+  bool h_zeroed = false;
+  bool directions[] = {DIR_MINUS, DIR_PLUS, DIR_MINUS};
+  unsigned int delays[] = {fast_delay_micros, fast_delay_micros, slow_delay_micros};
```
Comment (Nyarmith) [inline ClockAPI.h]: `unsigned int` and `unsigned` are equivalent types fyi

File: Clock.ino (hunk 2 above)
Comment (QuentinTorg) [reply Clock.ino]: I made the motor class and separated the zeroing of the gantry and hour hand

**Comment by QuentinTorg** [comment]
I made the motor class and separated the zeroing of the gantry and hour hand. the motors have an init function to assign pin modes.

**Review by Nyarmith — APPROVED** [review]
Nice job! :smile_cat:

**Pull request merged**